<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Arguments -->
  <arg name="map_file" default="$(find lemon_minibot_control)/map_data/Lemon_minibot.yaml"/>
  


  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

    <include file="$(find lemon_minibot_control)/launch/base_control.launch"/>

    <include file="$(find lemon_minibot_control)/launch/amcl.launch"/>

  <!-- rviz -->
  <!--<group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find turtlebot3_navigation)/rviz/turtlebot3_navigation.rviz"/>
  </group>-->
  
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
        <rosparam file="$(find lemon_minibot_control)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find lemon_minibot_control)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find lemon_minibot_control)/param/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find lemon_minibot_control)/param/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find lemon_minibot_control)/param/move_base_params.yaml" command="load" />
        <rosparam file="$(find lemon_minibot_control)/param/dwa_local_planner_params.yaml" command="load" />
        <remap from="cmd_vel" to="/cmd_vel"/>
        <remap from="odom" to="odom"/>
    </node>  
</launch>